lib/ts/IntrusiveHashMap.h

```diff
@@ -452,7 +452,7 @@ IntrusiveHashMap<H>::find(key_type key) -> iterator
   while (v != limit && !H::equal(key, H::key_of(v))) {
     v = H::next_ptr(v);
   }
-  return _list.iterator_for(v);
+  return _list.iterator_for(v == limit ? nullptr : v);
```

Comment:
We are saying that this is tha same as saying:
return v==limit ? _list.end() : _list.iterator_for(v);
this is the same as saying 
return v==limit ? this->end() : _list.iterator_for(v);

Reply:
This takes advantage of the fact that `iterator_for(nullptr)` yields the same result as `end()`.